### DISCUSSION TITLE
Explore using Kani for property proving (and Bolero to integrate this with fuzztests?)

### DISCUSSION THREAD

**Discussion by apoelstra** [rust-bitcoin/rust-bitcoin#1370] [Ideas]
See this blog post: https://model-checking.github.io/kani-verifier-blog/2022/10/27/using-kani-with-the-bolero-property-testing-framework.html

It's not clear to me that combinig Kani (proof checking) and Bolero (fuzzing with swappable backends), as described in the blog post, actually makes sense ... but we definitely should be writing property tests for a lot of our functionality.

**Comment by Kixunil** [comment]
Heh, was thinking about Kani too. Concept ACK.
